app/src/sharedTest/java/org/oppia/app/utility/DragViewAction.kt

```diff
@@ -0,0 +1,204 @@
+        object : DownMotionPerformer {
+          override fun perform(uiController: UiController, coordinates: FloatArray, precision: FloatArray): MotionEvent {
+            return performLongPress(uiController,coordinates,precision)
+          }
+        }
+
+      private fun performLongPress(uiController: UiController, coordinates: FloatArray, precision: FloatArray): MotionEvent {
+        val downEvent = MotionEvents.sendDown(uiController, coordinates, precision).down
+        // Duration before a press turns into a long press.
+        // Factor 1.5 is needed, otherwise a long press is not safely detected.
+        // See android.test.TouchUtils longClickView
+        val longPressTimeout = (ViewConfiguration.getLongPressTimeout() * 1.5f).toLong()
+        uiController.loopMainThreadForAtLeast(longPressTimeout)
+        return downEvent
+      }
+
+      override fun sendSwipe(uiController: UiController, startCoordinates: FloatArray, endCoordinates: FloatArray, precision: FloatArray): Swiper.Status {
+        return sendLinearDrag(uiController, downMotion, startCoordinates, endCoordinates, precision)
+      }
+
+      private fun sendLinearDrag(uiController: UiController, downMotion: DragViewAction.DownMotionPerformer, startCoordinates: FloatArray, endCoordinates: FloatArray, precision: FloatArray): Swiper.Status {
+        val steps = interpolate(startCoordinates,endCoordinates)
+        val delayBetweenMovements = DRAG_DURATION / steps.size
+        val downEvent = downMotion.perform(uiController, startCoordinates, precision) ?: return Swiper.Status.FAILURE
+        try {
+          steps.forEachIndexed { index, step ->
+            if (!MotionEvents.sendMovement(uiController, downEvent, step)) {
+              MotionEvents.sendCancel(uiController, downEvent)
+              return Swiper.Status.FAILURE
+            }
+            val desiredTime: Long = downEvent.downTime + delayBetweenMovements * index
+            val timeUntilDesired = desiredTime - SystemClock.uptimeMillis()
+            if (timeUntilDesired > 10) {
+              // If the wait time until the next event isn't long enough, skip the wait
+              // and execute the next event.
+              uiController.loopMainThreadForAtLeast(timeUntilDesired)
+            }
+          }
+          // Wait before sending up because some drag handling logic may discard move events
+          // that has been sent immediately before the up event. e.g. HandleView.
+          uiController.loopMainThreadForAtLeast(WAIT_BEFORE_SENDING_UP.toLong())
+          if (!MotionEvents.sendUp(uiController, downEvent, endCoordinates)) {
+            MotionEvents.sendCancel(uiController, downEvent)
+            return Swiper.Status.FAILURE
+          }
+        } finally {
+          downEvent.recycle()
+        }
+        return Swiper.Status.SUCCESS
+      }
+
+      private fun interpolate(start:FloatArray,end:FloatArray):Array<FloatArray>{
```

Comment:
I think this helper needs a KDoc as well to explain what it's doing--it may not be clear to everyone who reads it.